### ISSUE TITLE
Cognito Analytics are not being sent to Pinpoint when using Authenticator

### ISSUE THREAD

**Issue by ivan-kiselev** [aws-amplify/amplify-ui#4938]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Authenticator

### How is your app built?

Create React App

### What browsers are you seeing the problem on?

Firefox, Safari

### Which region are you seeing the problem in?

eu-central-1, us-east-1

### Please describe your bug.

User SignUp analytics don't show up in the PinpointAnalytics when using AWS Amplify Authenticator with configured Analytics block:

```javascript
Amplify.configure({
  Analytics: {
    Pinpoint: {
      bufferSize: 100,
      flushInterval: 30,
      flushSize: 100,
      resendLimit: 10,
      appId: "MY_ID",
      region: "eu-central-1",
    },
  },
  Auth: {
    Cognito: {
      userPoolClientId: "MY_ID",
      userPoolId: "MY_ID",
      signUpVerificationMethod: "link",
      loginWith: {
        username: true,
        email: true,
        phone: false,
      },
    },
  },
});
```

User sign up is performed successfully, the integration between the AWS Cognito and pinpoint analytics is configured in the console:

<img width="2030" alt="Screenshot 2024-01-21 at 16 28 38" src="https://github.com/aws-amplify/amplify-ui/assets/10528835/987a7b5f-3dda-4b08-9e58-04544d71f919">

### What's the expected behaviour?

When I sign up in my App, I want to see some activity in integrated with my Cognito client pinpoint project, but nothing happens.

Documentation [states](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-pinpoint-integration.html#cognito-user-pools-pinpoint-integration-client) that SignUp event from cognitoshall be reflected, though it also mentions vaguely that what shoyuld "include an `AnalyticsEndpointId` value in the `AnalyticsMetadata`  parameter of your API request", which I don't see happening if I inspect requests to Cognito from my UI when I press "sign up" button on Authenticator component. I only see essential for registration data:
<img width="76" alt="Screenshot 2024-01-21 at 16 35 37" src="https://github.com/aws-amplify/amplify-ui/assets/10528835/40b81c5c-94d2-4bdc-a2e0-f243ca7d399c">

### Help us reproduce the bug!

Set up cognito user pool, cognito client and integrate the client with pinpoint analytics.

Use this client for web-interface based on the following snippet

### Code Snippet

```typescript
// Put your code below this line.
Amplify.configure({
  Analytics: {
    Pinpoint: {
      bufferSize: 100,
      flushInterval: 30,
      flushSize: 100,
      resendLimit: 10,
      appId: "...",
      region: "eu-central-1",
    },
  },
  Auth: {
    Cognito: {
      userPoolClientId: "...",
      userPoolId: "...",
      signUpVerificationMethod: "link",
      loginWith: {
        username: true,
        email: true,
        phone: false,
      },
    },
  },
});
```
And Cogntio user pool with a secretless client integrated with Pinpoint app in the same region, and authenticator:

```html
        <Authenticator signUpAttributes={["email"]}>
          {({ signOut, user }) => (
            <View>
              <Heading color="pieCrust.700">Hello {user?.username}</Heading>
              <Text textAlign="center">Welcome to our application!</Text>
              <Button onClick={signOut}>Sign out</Button>
            </View>
          )}
        </Authenticator>

```

### Additional information and screenshots

I generally see AnalyticsMetadata in the [corresponding library](https://github.com/aws-amplify/amplify-js/blob/551a58c52347482896f72acfd363984913434cb4/packages/auth/src/providers/cognito/utils/clients/CognitoIdentityProvider/types.ts#L1502), but I am not sure if I need to do something additionally myself to make the client propagate this field.

I am not really good with frontend, trying me best here, sorry if I am missing something absolutely obvious.

**Comment by calebpollman** [comment]
@ivan-kiselev Going to transfer this issue over to our sibling team Amplify JS to see if they can unblock you with the Pinpoint/Cognito integration piece as analytic recording is not handled by the `Authenticator` without customization

@cwomack Based on the provided info above (see [cognito docs](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-pinpoint-integration.html#cognito-user-pools-pinpoint-integration-client)) is this something we can help with?
